instrumentation-tools/src/api_surface.py

```diff
@@ -61,43 +44,46 @@ def sanitize_prefix(previous_indent: int,
 
 
 def resolve_type_name(type_definition) -> str:
-    return type_definition.strip().split(EXTEND_SEPARATOR)[0].strip().split(DOT)[-1]
+    return type_definition.strip().split(EXTEND_SEPARATOR)[0].strip()
 
 
 class ApiSurface:
     file_path: str
-    extra_ignored_types_reg_ex: re
+    ignored_types: [str]
 
     def __init__(self, file_path: str, ignored_types: [str] = None):
         self.file_path = file_path
-        if ignored_types:
-            self.extra_ignored_types_reg_ex = re.compile(f"^{'|'.join(ignored_types)}", re.IGNORECASE)
-        else:
-            self.extra_ignored_types_reg_ex = None
+        self.ignored_types = ignored_types
 
     def fetch_testable_methods(self) -> set:
         to_return = set()
         with open(self.file_path, 'r') as file:
             rows = file.readlines()
-        prefix = ""
+        stack = ["", "", "", "", "", ""] # we don't expect nested types with more than 5 levels
         # We will need this prefix which holds also the attributes of the type (e.g. "companion", "deprecated", "open")
         # to be able to ignore some APIs.
-        prefix_with_type_attributes = ""
+        type_attributes = ""
+        type = ""
 
-        current_indent = 0
         for row in rows:
             is_type_matcher = TYPE_REGEX.match(row)
+            is_testable_api = TESTABLE_API_REGEX.match(row)
```

Comment:
This code is quite complex, maybe a good idea would be (something that I should have added when I first wrote this) is to add some unit tests. Maybe we can open a new task for this and we can add those next sprint (will be glad to handle that).

Reply:
What's complex about it ?

Reply:
well it much more easier to update it later with unit tests in place.